pallets/ddc-clusters/src/lib.rs

```diff
@@ -835,6 +835,15 @@ pub mod pallet {
 	}
 
 	impl<T: Config> ClusterManager<T> for Pallet<T> {
+		fn get_all_clusters() -> Result<Vec<ClusterId>, DispatchError> {
+			let mut cluster_ids = Vec::new();
+
+			for cluster_id in <Clusters<T>>::iter_keys() {
```

Comment:
Stuff like this is applicable only in the scope of the demo. Having unfounded iterators in runtime code / smart contract logic is a security issue and anti-pattern.

Reply:
- check for status Activated
- number of clusters is in low dozens so shouldn't be an issue